loader/src/sources/vts/geo.js

```diff
@@ -0,0 +1,164 @@
+      }
+      return data.name;
+    },
+  },
+};
+
+function warn(message, context) {
+  console.warn(`VTS Geo: ${message}`, context);
+  Sentry.captureMessage(message, Sentry.Severity.Info);
+}
+
+function error(message, context) {
+  console.error(`VTS Geo: ${message}`, context);
+  Sentry.captureMessage(message, Sentry.Severity.Error);
+}
+
+async function getStores() {
+  try {
+    const response = await got({
+      url: `https://univaf-data-snapshots.s3.us-west-2.amazonaws.com/vts/vts-final-output-locations.geojson`,
+    });
+    return JSON.parse(response.body);
+  } catch (e) {
+    error(`Error fetching stored Vaccine the States data`, e);
+    return [];
+  }
+}
+
+/**
+ * Filter out some stores with bad data.
+ * @param {any} store
+ * @returns {boolean}
+ */
+function hasUsefulData(store) {
+  return (
+    store.properties.concordances?.length &&
+    store.geometry?.coordinates &&
+    store.properties.provider?.name &&
+    store.properties.provider?.name in dataProviders
+  );
+}
+
+function formatStore(store) {
+  const data = store.properties;
+
+  let result;
+  Sentry.withScope((scope) => {
+    const dataProvider = dataProviders[data.provider.name];
+    if (!dataProvider) {
+      error(`Unexpected provider name ${data.provider.name}`);
+      return null;
+    }
+
+    const provider = dataProvider.provider;
+    const name = dataProvider.getStoreName(data);
+
+    scope.setContext("location", {
+      id: store.id,
+      url: data.vts_url,
+      name: name,
+      provider: provider,
+    });
+
+    function splitConcordance(concordance) {
+      const colon = concordance.indexOf(":");
+      return [
+        concordance.substring(0, colon),
+        concordance.substring(colon + 1),
+      ];
+    }
+
+    function filterConcordancePairs(pairs) {
+      return pairs
+        .filter(([systemCode, value]) => {
+          const system = systemsToSend[systemCode];
+          if (!system) {
+            return null;
+          }
+
+          if (system == "rite_aid" && value.match(/^1\d{5}$/)) {
+            // Stores like Rite Aid in this data set may have a store number like 105612.
+            // IDs with that shape (6 digits, starting with '1') are likely misparsed from VTrckS.
+            return null;
+          }
+
+          return [system, value];
+        })
+        .filter(Boolean);
+    }
+
+    const systemsToSend = {
+      cvs: "cvs",
+      rite_aid: "rite_aid",
+      vaccinespotter_org: "vaccinespotter",
+      vaccinefinder_org: "vaccines_gov",
+      vaccinefinder: "vaccines_gov",
+    };
+
+    const concordances = data.concordances.map(splitConcordance);
+    const externalIds = filterConcordancePairs(concordances);
+    const univafPair = concordances.filter((v) => v[0] == "getmyvax_org")[0];
+
+    if (!externalIds.length) {
+      return null;
+    }
+
+    result = {
+      name,
+      provider,
+      id: univafPair ? univafPair[1] : undefined,
```

Comment:
I feel a little iffy about actually using this mapping because of issues like #212. Did you do any checks to see how good a match this is likely to be or how often a location has more than one of our IDs or one of our IDs is used across multiple locations?

Reply:
In this data set there's only one duplicated `getmyvax_org` id. Here are the stores:

```json
{
  "type": "Feature",
  "id": "lxwcf",
  "properties": {
    "name": "MEDICINE SHOPPE PHARMACY",
    "provider": {
      "name": "CVS",
      "provider_type": "Pharmacy",
      "vaccine_info_url": "https://www.cvs.com/immunizations/covid-19-vaccine"
    },
    "state": "MO",
    "location_type": "Unknown",
    "phone_number": null,
    "full_address": "149 POND FORT TRL\nLAKE ST LOUIS, MO 63367",
    "city": "LAKE ST LOUIS",
    "county": "St. Charles",
    "zip_code": "63367",
    "hours": {
      "unstructured": null,
      "structured": null
    },
    "website": null,
    "vaccines_offered": null,
    "accepts_appointments": null,
    "accepts_walkins": null,
    "concordances": [
      "mo_arcgis:2a9658e6-f87c-4299-929d-cccaba3b879e",
      "placekey:222-222@5pk-qs2-4gk",
      "_tag_provider:cvs",
      "placekey:zzw-224@5pk-qrx-6c5",
      "vtrcks:CV1018053",
      "getmyvax_org:54ff60f9-0c5f-4b13-920c-c245fcf6424c",
      "cvs:18053"
    ],
    "last_verified_by_vts": null,
    "vts_url": "https://www.vaccinatethestates.com/?lng=-90.78767&lat=38.76361#lxwcf"
  },
  "geometry": {
    "type": "Point",
    "coordinates": [
      -90.78767,
      38.76361
    ]
  }
}
{
  "type": "Feature",
  "id": "lcdtry",
  "properties": {
    "name": "CVS #18053 (inside Schnuck's)",
    "provider": {
      "name": "CVS",
      "provider_type": "Pharmacy",
      "vaccine_info_url": "https://www.cvs.com/immunizations/covid-19-vaccine"
    },
    "state": "MO",
    "location_type": "Pharmacy",
    "phone_number": "(888) 607-4287",
    "full_address": "7909 Highway N. Dardenne Prairie, MO 63368",
    "city": "Dardenne Prairie",
    "county": "St. Charles",
    "zip_code": "63368",
    "hours": {
      "unstructured": "",
      "structured": null
    },
    "website": "https://www.cvs.com/vaccine/intake/store/cvd-schedule",
    "vaccines_offered": null,
    "accepts_appointments": true,
    "accepts_walkins": false,
    "concordances": [
      "_tag_provider:cvs",
      "placekey:zzw-224@5pk-qrx-6c5",
      "vtrcks:CV1018053",
      "getmyvax_org:54ff60f9-0c5f-4b13-920c-c245fcf6424c",
      "cvs:18053"
    ],
    "last_verified_by_vts": null,
    "vts_url": "https://www.vaccinatethestates.com/?lng=-90.76839&lat=38.76884#lcdtry"
  },
  "geometry": {
    "type": "Point",
    "coordinates": [
      -90.76839,
      38.76884
    ]
  }
}
```

I don't know enough here to say if they should or should not both be pointed at this location: https://getmyvax.org/api/edge/locations/54ff60f9-0c5f-4b13-920c-c245fcf6424c

But I think for the purpose of updating geolocations, either of the longitude/latitudes here are probably fine. Lemme know what you think, though.

Reply:
Well, that first one clearly is *very* wrong — it’s in a different town altogether and about a mile and a half down the road from the second one (which exactly matches the location with the given ID in our DB). The first one isn’t even a CVS (A more local franchise: https://www.lslmedshoppe.com/).

That also means the `_tag_provider`, `vtrcks`, `getmyvax_org`, and `cvs` concordances on the first one are wrong. I’d bet one of the `placekey` ones is correct and the other is not. :\

Also fun: It looks like vaccines.gov has two separate entries for the first place ([4d0eed17-693f-45d7-bc2a-9f0612e6deb9](https://www.vaccines.gov/provider/?id=4d0eed17-693f-45d7-bc2a-9f0612e6deb9) and [b16487fd-4da1-4a13-bf28-ffd50a529d64](https://www.vaccines.gov/provider/?id=b16487fd-4da1-4a13-bf28-ffd50a529d64)) which we’ll have to dedupe in or after merging #312.

Anyway, this just keeps me skeptical of using our IDs from their concordances list. Since we are only looking at CVS and Rite Aid for now, the actual `cvs` or `rite_aid` concordances *seem* like better indicators to key off of (although they are obviously wrong in the example you found here, too).

Reply:
Cool, will just remove the univaf ID sending altogether. Fingers crossed most of the `cvs` and `rite_aid` concordances are good?